pkg/controller/datavolume-controller.go

```diff
@@ -367,6 +367,14 @@ func (r *DatavolumeReconciler) getSnapshotClassForSmartClone(dataVolume *cdiv1.D
 		return "", errors.New("no source PVC provided")
 	}
 
+	preferHostAssisted, err := preferHostAssistedCopy(r.client)
```

Comment:
This seems like the wrong spot to check for this. The function is called getSnapshotClassForSmartClone, so we are interested in getting the snapshot class if it exists, regardless of the preference. I would check in whatever function calls getSnapshotClassForSmartClone similar to this:
```go
preferredStrategy, err := c.getPreferrredCloneStrategy()
if err != nil {
  return err
}
if preferredStrategy == "smart" {
   // whatever was calling getSnapshotClassForSmartClone
}
```